ansible/server-setup/roles/security/tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Install ufw
+  apt:
+    name: ufw
+    state: present
+    update_cache: yes
+
+- name: Install fail2ban
+  apt:
+    name: fail2ban
+    state: present
+    update_cache: yes
+
+- name: Restart fail2ban
+  service:
+    name: fail2ban
+    state: restarted
+  changed_when: false
+
+- name: Open ssh port
+  ufw:
+    rule: allow
+    port: "{{ ssh_port }}"
+    proto: tcp
+
+- name: Open http port
+  ufw:
+    rule: allow
+    port: '80'
+    proto: tcp
+
+- name: Open https port
+  ufw:
+    rule: allow
+    port: '443'
+    proto: tcp
+
+- name: Deny port 53
+  ufw:
+    rule: deny
+    port: 53
+    proto: tcp
+
+- name: Disable root login over SSH
+  lineinfile:
+    dest: '/etc/ssh/sshd_config'
+    regexp: '^PermitRootLogin'
+    line: 'PermitRootLogin no'
+    state: present
+  when: disable_root|default(true)|bool == true
+
+- name: Disable password login
+  lineinfile:
+    dest: '/etc/ssh/sshd_config'
+    regexp: '^PasswordAuthentication'
+    line: 'PasswordAuthentication no'
+    state: present
+
+- name: Restart SSH
+  service:
+    name: ssh
+    state: restarted
+  changed_when: false
+
+- name: Enable firewall
```

Comment:
Shouldn't we enable firewall first and then set rules after ?

Reply:
No, the firewall can be enabled as a last step.